presto-cli/src/main/java/io/prestosql/cli/Console.java

```diff
@@ -127,6 +129,22 @@ public boolean run()
             }
         }
 
+        // Read queries from stdin
+        if (!hasQuery && !detectRealTerminal()) {
```

Comment:
I guess this is fine. But other option to consider is also allowing passing `-` as a filename